### PR TITLE
Fix #1773 - drawBoldTextInBrightColors is not taken into account with DOM renderer

### LINKED ISSUE #1773: drawBoldTextInBrightColors is not taken into account with DOM renderer
## Details
- Browser and browser version: Chrome 70
- OS version: Mac OS X
- xterm.js version: 3.8.0

### Steps to reproduce

1. Set `drawBoldTextInBrightColors` to `false` and `rendererType` to `dom`
2. Bold characters are still rendered brighter

It works as expected with the canvas renderer.

### PR DESCRIPTION
**Pull request by nikonso**
Fix #1773 by taking into account the terminal options

### CODE REVIEW

File: src/renderer/dom/DomRendererRowFactory.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export const CURSOR_STYLE_UNDERLINE_CLASS = 'xterm-cursor-underline';
 
 export class DomRendererRowFactory {
   constructor(
+    private _terminal: ITerminal,
```
Comment (Tyriar) [inline src/renderer/dom/DomRendererRowFactory.ts]: I've been slowly trying to break this pattern of passing `ITerminal` around everywhere (https://github.com/xtermjs/xterm.js/issues/1627), I think passing in `ITerminalOptions` here would be preferable as we never reassign that on `Terminal`.

**Comment by nikonso** [comment]
Makes sense, I'll make the changes Sunday when I'm back home ??

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by Tyriar** [comment]
Sounds good, thanks a bunch as always 😃

**Comment by nikonso** [comment]
Circular dependency removed, and verified that it still works!

**Pull request merged**